Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -10,10 +14,42 @@ namespace JokesOnYou.Web.Api.Services
     public class JokesService : IJokesService
     {
         private readonly IJokesRepository _jokesRepo;
+        private readonly IUserRepository _userRepository;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
 
-        public JokesService(IJokesRepository jokesRepo)
+        public JokesService(IJokesRepository jokesRepo, IUserRepository userRepository, IUnitOfWork unitOfWork, IMapper mapper)
         {
             _jokesRepo = jokesRepo;
+            _mapper = mapper;
+            _userRepository = userRepository;
+            _unitOfWork = unitOfWork;
+        }
+        public async Task<JokeReplyDto> CreateJokeAsync(JokeCreateDto jokeCreateDto, string userId)
+        {
+
+            var user = await _userRepository.GetUserAsync(userId);
+            var joke = _mapper.Map<Joke>(jokeCreateDto);
+            joke.UploadDate = DateTime.Now;
+            joke.Author = user;
+
+            await _jokesRepo.CreateJokeAsync(joke);
+            var saved = await _unitOfWork.SaveAsync();
+            if (!saved)
+            {
+                throw new AppException("Error with saving the Joke.");
+            }
+
+            var jokeReplyDto = _mapper.Map<JokeReplyDto>(joke);
```

Comment:
Should we Get the Joke from the database to double check if its there?
Then return that instance ?

Reply:
Whats the reason for returning a joke here considering we are creating it? So if the other joke doesnt reach the db then we never get here since we make sure the stuff were saved.  Im not sure what kinda edge cases there are here @PoisnFang

Reply:
The reason i would return the joke here is to show it to the person that created it. just like you would show a Registered user its account after registering. but it could be done differently of course.

Reply:
Usually its recommended that the client just does a Get request after the Post. If we wanna follow REST a bit more and less chance of errors